tests/stores/directory/test_local_directory.py

```diff
@@ -111,30 +111,169 @@ def test_add_file_does_not_exist():
+        current_source_pathlike = test_local_dir_store.get_current_source_pathlike()
+        tmp_dir_absolute = Path(tmp_dir).absolute()
+        assert current_source_pathlike == str(tmp_dir_absolute)
+
+
+def test_has_lone_file_matching():
+    """
+    Checks that a LocalDirectoryStore can run the has_lone_file_matching()
+    function against its local directory and successfully returns True if 
+    the directory has only one file matching the given pattern.
+    """
+
+    test_path = Path("tests/test_cases/test_local_store/local_directory_folders/local_directory_lone_file")
+    test_local_directory_store = LocalDirectoryStore(test_path)
+
+    assert test_local_directory_store.has_lone_file_matching(".json")
+
+def test_has_lone_file_matching_none():
+    """
+    Checks that a LocalDirectoryStore can run the has_lone_file_matching()
+    function against its local directory and successfully returns False if 
+    the directory has zero files that match the given pattern.
+    """
+
+    test_path = Path("tests/test_cases/test_local_store/local_directory_folders/local_directory_lone_file")
+    test_local_directory_store = LocalDirectoryStore(test_path)
+
+    assert not test_local_directory_store.has_lone_file_matching(".py")
+
+
+def test_has_lone_file_matching_multiple():
+    """
+    Checks that a LocalDirectoryStore can run the has_lone_file_matching()
+    function against its local directory and successfully returns False if 
+    the directory has more than one file that matches the given pattern.
+    """
+
+    test_path = Path("tests/test_cases/test_local_store/local_directory_folders/local_directory_multiple_file")
+    test_local_directory_store = LocalDirectoryStore(test_path)
+
+    with pytest.raises(FileNotFoundError) as err:
+        test_local_directory_store.has_lone_file_matching(".json")
+    assert "More than 1 file found that matches the regex pattern '.json' in directory tests/test_cases/test_local_store/local_directory_folders/local_directory_multiple_file. Matching: ['local_directory2.json', 'local_directory1.json']" == str(err.value)
+
+
+def test_save_lone_file_destination():
+    """
+    Checks that a LocalDirectoryStore can retrieve a lone matching file 
+    from a given pattern in its local path directory, and save that file 
+    in a given destination directory.
+    """
+
+    destination_dir = Path("tests/test_cases/test_local_store/local_directory_folders/local_directory_destination")
+    destination_dir.mkdir(exist_ok=True)
+
+    test_path = Path("tests/test_cases/test_local_store/local_directory_folders/local_directory_lone_file")
+    test_local_directory_store = LocalDirectoryStore(test_path)
+
+    test_local_directory_store.save_lone_file_matching(".json", destination_dir)
+    test_result_file = destination_dir / "local_directory_test.json"
+
+    assert test_result_file.exists()
+
+    # Delete the saved file and directory after test is completed.
+    os.remove(test_result_file)
```

Comment:
Instead of creating and then removing the file/directory, you could use `with TemporaryDirectory...` - have a look at the tests for `add_file()`. Same for other tests using this approach.